hw3/01-characters.js

```diff
@@ -1,4 +1,75 @@
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
+const sectionToAddTo = document.querySelector('section');
+sectionToAddTo.classList.add('container-fluid', 'row');
 
-fetch(url);
+// fetch the data from the API
+fetch(url)
+  .then((data) => {
+    return data.json();
+  }) // return the json of the data
+  .then((characters) => {
+    displayCharacters(characters, sectionToAddTo);
+  }) // call fn to display it (append to the given element)
+  .catch((error) => {
+    console.log(error);
+  }) // catch any errors and throw them at the console
+  .finally(() => {
+    console.log('API call completed');
+  }); // why not have a finally? Sounds like a decent idea!
+
+// remove all the child elements for a given element
+function removeAllChildren(element) {
+  while (element.firstChild !== null) {
+    element.removeChild(element.firstChild);
+  }
+}
+
+function displayCharacters(characters, location) {
+  removeAllChildren(location);
+
+  characters.forEach((character) => {
+    // wrap the column so that we can use grid for display of it's items
+    let colWrapper = document.createElement('div');
+    colWrapper.classList.add('col-6', 'col-lg-3', 'col-wrapper');
+
+    // create a card element to contain each character
+    let card = document.createElement('div');
+    card.classList.add(
```

Comment:
I never looked at the documentation for add() and didn't realize you could provide a list of classes. Definitely cleans up the code!

Reply:
Oh yeah! I realized this when looking over some code from my partner in the group project. Def make the code look cleaner